### PR TITLE
XGB Regressor

### PR DESCRIPTION
**Pull request by Moddy2024**
> ### Describe your change:
> * [x]  Add an algorithm?
> * [ ]  Fix a bug or typo in an existing algorithm?
> * [ ]  Documentation change?
> 
> ### Checklist:
> * [x]  I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
> * [x]  This pull request is all my own work -- I have not plagiarized.
> * [x]  I know that pull requests will not be merged if they fail the automated tests.
> * [x]  This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
> * [x]  All new Python files are placed inside an existing directory.
> * [x]  All filenames are in all lowercase characters with no spaces or dashes.
> * [x]  All functions and variable names follow Python naming conventions.
> * [x]  All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
> * [x]  All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
> * [x]  All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
> * [x]  If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Moddy2024** [comment]
@cclauss  Can you please review this I hope this will be merged as all the tests are passing.

File: machine_learning/xgboostregressor.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from xgboost import XGBRegressor
+
+
+def dataset(datatype: dict) -> tuple:
+    # Split dataset into train and test data
+    x = (datatype["data"], datatype["target"])  # features
+    return x
+
+
+def xgboost(features: list, target: list, test_features: list) -> list:
```
[the remark this answers is not in the document]
Comment (Moddy2024) [reply machine_learning/xgboostregressor.py:14]: Already Provided

File: machine_learning/xgboostregressor.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# XGBoost Regressor Example
+from sklearn.datasets import load_boston
+from sklearn.metrics import mean_absolute_error, mean_squared_error
+from sklearn.model_selection import train_test_split
+from xgboost import XGBRegressor
+
+
+def dataset(datatype: dict) -> tuple:
```
[the remark this answers is not in the document]
Comment (Moddy2024) [reply machine_learning/xgboostregressor.py:8]: Already Provided

File: machine_learning/xgboostregressor.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+def xgboost(features: list, target: list, test_features: list) -> list:
+    xgb = XGBRegressor()
+    xgb.fit(features, target)
+    # Predict target for test data
+    predictions = xgb.predict(test_features)
+    predictions = predictions.reshape(len(predictions), 1)
+    return predictions
+
+
+def main() -> None:
```
[the remark this answers is not in the document]
Comment (Moddy2024) [reply machine_learning/xgboostregressor.py:23]: Already Provided

**Comment by Moddy2024** [comment]
@AkarshSatija  @Erfaniaa  Hello, Can anyone Please review this. Thank you.

**Comment by Moddy2024** [comment]
@AkarshSatija @Erfaniaa Can anyone review this please whenever you get time as it has been 6 days since I created the PR.
Thank you

**Comment by Moddy2024** [comment]
@cclauss Hello can you please review this whenever you get time.
Thank You

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
These are ***how-to-use*** XGBoost demonstrations rather than algorithms as defined in [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
Please make sure that future pull requests focus on delivering an algorithm.

**Comment by Moddy2024** [comment]
> These are _**how-to-use**_ XGBoost demonstrations rather than algorithms as defined in [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md). Please make sure that future pull requests focus on delivering an algorithm.

Yes Sir I will make sure of that.
Thanks for helping me and teaching me on the way.

**Pull request merged**